File: E-Tour/src/main/java/com/example/demo/entities/Cancellation.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "cancellation")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Cancellation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cancellation_id")
    @EqualsAndHashCode.Include
    private Integer cancellationId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "booking_id")
    @ToString.Exclude
    private Booking booking;

    @Column(name = "cancellation_date")
    private LocalDateTime cancellationDate;

    @Column(name = "reason", length = 255)
    private String reason;

    @Column(name = "refund_amount", precision = 12, scale = 2)
    private BigDecimal refundAmount;

    @Column(name = "refund_status", length = 30)
    private String refundStatus;

    @Column(name = "remarks", length = 255)
    private String remarks;
}

File: ETour-DotNet/Models/Cancellation.cs
namespace ETour.Api.Models;

public class Cancellation
{
    public int CancellationId { get; set; }
    public int? BookingId { get; set; }
    public Booking Booking { get; set; }
    public DateTime? CancellationDate { get; set; }
    public string Reason { get; set; }
    public decimal? RefundAmount { get; set; }
    public string RefundStatus { get; set; }
    public string Remarks { get; set; }
}
